crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -11,9 +11,33 @@ where
 {
     pub(crate) async fn block_by_hash(
         &self,
-        _hash: H256,
+        hash: H256,
         _full: bool,
     ) -> EthResult<Option<RichBlock>> {
-        todo!()
+        let block = self.client().block(BlockId::Hash(hash.0.into()))?;
+        if let Some(_block) = block {
+            // TODO: GET TD FOR BLOCK - needs block provider? or header provider?
```

Comment:
Fix by introducing `Client: HeaderProvider + <..>` and use the `header_td` method